ui/components/app/wallet-overview/coin-buttons.tsx

```diff
@@ -355,6 +364,39 @@ const CoinButtons = ({
     defaultSwapsToken,
   ]);
 
+  const handleReceiveOnClick = useCallback(() => {
+    trace({ name: TraceName.ReceiveModal });
+    trackEvent({
+      event: MetaMetricsEventName.NavReceiveButtonClicked,
+      category: MetaMetricsEventCategory.Navigation,
+      properties: {
+        text: 'Receive',
+        location: trackingLocation,
+        // TODO: Fix in https://github.com/MetaMask/metamask-extension/issues/31860
+        // eslint-disable-next-line @typescript-eslint/naming-convention
+        chain_id: chainId,
+      },
+    });
+
+    // Check if multichain accounts feature is enabled and we have a selected account group
+    if (isMultichainAccountsState1Enabled && selectedAccountGroup) {
+      // Navigate to the multichain address list page with receive source
+      history.push(
+        `${MULTICHAIN_ACCOUNT_ADDRESS_LIST_PAGE_ROUTE}/${encodeURIComponent(selectedAccountGroup)}?source=receive`,
+      );
+    } else {
+      // Show the traditional receive modal
+      setShowReceiveModal(true);
+    }
+  }, [
+    isMultichainAccountsState1Enabled,
+    selectedAccountGroup,
+    history,
+    trackEvent,
+    trackingLocation,
+    chainId,
+  ]);
```

Comment:
If `isMultichainAccountsState1Enabled` is enabled we will see the address list page navigation from the home page Receive button


https://github.com/user-attachments/assets/46f9a8da-39f8-48f7-acb6-e7b497b6b1ac

If it is not enabled see normal QR code modal